content/docs/user-guide/visualization/environments/terrain/terrain-developer-guide/system-usage.md

```diff
@@ -0,0 +1,85 @@
+
+### Terrain world grids
+
+The underlying terrain data can be at any resolution, including "infinite" resolution for procedural data like random number generators, and the data resolution can vary by position across different Terrain Layer Spawners. The terrain system defines separate conceptual world grids for height data and surface data so that all of the input data is normalized into a single output resolution, which is defined by the grid spacing.
+
+The grid spacing is controlled by the `Height Query Resolution` and `Surface Data Query Resolution` settings. A query resolution of 1 meter means that across the entire world, the terrain will query the underlying data at a consistent 1 meter spacing, regardless of what resolution the underlying data is at.
+
+However, the terrain system itself can be queried at any position, not just on the conceptual grids. The `Sampler` type controls exactly _how_ to query the underlying data in relation to these grids.
+
+The `Default` / `Bilinear` Sampler setting uses the query resolution to sample the four grid corners that surround the requested position, and then it performs bilinear filtering to create the resulting output value. For example, in the illustration below, the underlying height data has different values at every 0.25 meter interval. The chart below it shows what happens when querying this data at position 0.75 with different Height Query Resolution settings.
+
+```goat
+40  o                           o
+30  |                           |
+20  |      o      o             |
+10  |      |      |      o      |
+0   +------+------+------+------+
+   0.00   0.25   0.50   0.75   1.00
+```
+
+| Height Query Resolution | Result | Description |
+| - | - | - |
+| 1.0 meters | 40 | Querying at 0.75 will fetch the underlying data at 0.00 (40) and 1.00 (40) and interpolate at 75%. |
+| 0.5 meters | 30 | Querying at 0.75 will fetch the underlying data at 0.50 (20) and 1.00 (40) and interpolate at 50%. |
+| 0.25 meters | 10 | Querying at 0.75 will fetch the underlying data at 0.75 (10) and 1.00 (40) and interpolate at 0%. |
+
+The `Default` Sampler setting is the one that should be used by most systems when querying. However, there are two other sampling methods that are available for specialized use.
+
+The `Clamp` Sampler setting will take the requested query position and 'clamp' it down to the terrain grid before requesting the value. Where the `Bilinear` sampler produces an infinite range of interpolated values for all the positions between grid points, the `Clamp` sampler will always produce the same value for every position between grid points. Because this only fetches one data value, not four, it can run up to 4x faster than the `Default` sampler, but the values returned will only match the rendering and physics representations when queried directly on the grid points. For that reason, this setting should only be used when querying directly on terrain grid points.
```

Comment:
Small suggestion to recall that there are "four" positions here because it helps quickly answer "why is it 4x faster?"

```suggestion
The `Clamp` Sampler setting takes the requested query position and "clamps" it down to the terrain grid before requesting the value. Whereas the `Bilinear` sampler produces an infinite range of interpolated values for all four positions between the grid points, the `Clamp` sampler always produces the same value for each position. Because this only fetches one data value, not four, it can run up to 4x faster than the `Default` sampler, but the returned values only match the rendering and physics representations when they are queried directly on the grid points. For that reason, this setting should only be used when querying directly on terrain grid points.
```

Reply:
This might need better clarification, "four" isn't a correct change here. 

Suppose you have four grid points like this where the heights are 5 at X=1, and 6 at X=2:
```
2 +  *5 *6
  |
1 +  *5 *6
  |    
0 +--+--+
   0  1  2
```

What this section is trying to say is that you can query an infinite number of positions between those grid points and get back an infinite number of results. For example, a query of (1.5, 1) gives you a height of 5.5. a query of (1.5001, 1) gives you a height of 5.5001. And so on. That's what "all the positions" is supposed to mean in this context.

In contrast, when using using 'Clamp', a query of (1.5, 1) gives you a height of 5. A query of (1.5001, 1) gives you a height of 5. A query of (1.999, 1) gives you a height of 5. But a query of (2, 1) gives you a height of 6.

Open to suggestions on how to make this more clear...

Reply:
I've reworked this suggestion and included (hopefully) better diagrams. What do you think?